specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-03-01-preview/costallocation.json

```diff
@@ -0,0 +1,689 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "CostManagementClient",
+    "version": "2020-03-01-preview"
+  },
+  "tags": [
+    {
+      "name": "CostAllocationRules",
+      "description": "An API for creating and managing cost allocation rules within a billing account or enterprise enrollment. These rules can be used to reallocate cost between resources, filtered by Resource Group, Subscription, or Tag."
+    }
+  ],
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/costAllocationRules": {
+      "get": {
+        "tags": [
+          "CostAllocationRules"
+        ],
+        "operationId": "CostAllocationRules_List",
+        "description": "Get the list of all cost allocation rules for a billing account or enterprise enrollment.",
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "CostAllocationRulesList": {
+            "$ref": "./examples/CostAllocationRulesList.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/billingAccountIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of cost allocation rules for the billing account or enterprise enrollment.",
+            "schema": {
+              "$ref": "#/definitions/CostAllocationRuleList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountId}/providers/Microsoft.CostManagement/costAllocationRules/checkNameAvailability": {
```

Comment:
Move the action checkNameAvailability directly under the provider, and then the resource type can be taken in the request body, this will let you use the same method for other resource types where you want to provide this functionality of check name. And it would be compliant with ARM Resource provuder contract - 
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#check-name-availability-requests

Reply:
Is this change required? We already reviewed having the field this way previously (https://github.com/Azure/azure-rest-api-specs-pr/pull/1152) and now have a number of dependencies based on this. Is this a recent change? I thought we configured per recommendations when we implemented it and did not specify resource type.